rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/SecureCxfClientFactoryBuilder.java

```diff
@@ -0,0 +1,351 @@
+                SecuritySettingsService service = bundleContext.getService(serviceReference);
+                if (service != null) {
+                    return service;
+                }
+            }
+        }
+        throw new SecurityServiceException("Could not get SecuritySettingsService");
+    }
+
+    private STSClientConfiguration getStsClientConfiguration() throws SecurityServiceException {
+        BundleContext bundleContext = FrameworkUtil.getBundle(this.getClass()).getBundleContext();
+        if (bundleContext != null) {
+            ServiceReference<STSClientConfiguration> serviceReference = bundleContext
+                    .getServiceReference(STSClientConfiguration.class);
+            if (serviceReference != null) {
+                STSClientConfiguration service = bundleContext.getService(serviceReference);
+                if (service != null) {
+                    return service;
+                }
+            }
+        }
+        throw new SecurityServiceException("Could not get STSClientConfiguration");
+    }
+
+    /**
+     * @see SecureCxfClientFactoryBuilder#buildFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass)
+            throws SecurityServiceException {
+        return buildFactory(endpointUrl, interfaceClass, null, null);
+    }
+
+    /**
+     * @see SecureCxfClientFactoryBuilder#buildFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass,
+            String username, String password) throws SecurityServiceException {
+        return buildFactory(endpointUrl, interfaceClass, username, password, null, false);
+    }
+
+    /**
+     * Creates a factory that will return security-aware clients.
+     *
+     * @see SecureCxfClientFactoryBuilder.SecureCxfClientFactory
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass,
+            String username, String password, List<?> providers, boolean disableCnCheck)
+            throws SecurityServiceException {
+        if (StringUtils.isEmpty(endpointUrl) || interfaceClass == null) {
+            throw new IllegalArgumentException(
+                    "Called without a valid URL, will not be able to connect.");
+        }
+
+        JAXRSClientFactoryBean jaxrsClientFactoryBean = new JAXRSClientFactoryBean();
+        jaxrsClientFactoryBean.setServiceClass(interfaceClass);
+        jaxrsClientFactoryBean.setAddress(endpointUrl);
+        jaxrsClientFactoryBean.setClassLoader(interfaceClass.getClassLoader());
+        jaxrsClientFactoryBean.getInInterceptors().add(new LoggingInInterceptor());
+        jaxrsClientFactoryBean.getOutInterceptors().add(new LoggingOutInterceptor());
+
+        if (!CollectionUtils.isEmpty(providers)) {
```

Comment:
Good catch. I didn't see the different package. The 'real' commons collection is exposed on the osgi runtime so you wouldn't need to embed it.